### PR TITLE
DOC: whatsnew fixes

### PR DESCRIPTION
**Pull request by jschendel**
Some small fixes I noticed reading over the whatsnew.

### CODE REVIEW

File: doc/source/whatsnew/v1.0.0.rst [hunk 1]
```diff
@@ -28,7 +28,7 @@ See :ref:`policies.version` for more.
 Enhancements
 ~~~~~~~~~~~~
 
-.. _whatsnew_100.NA:
+.. _whatsnew_1000.NA:
```
Comment (jreback) [inline doc/source/whatsnew/v1.0.0.rst]: why is this changing?

File: doc/source/whatsnew/v1.0.0.rst [hunk 2]
```diff
@@ -712,7 +712,7 @@ or ``matplotlib.Axes.plot``. See :ref:`plotting.formatters` for more.
 - Removed :meth:`DataFrame.from_items` (:issue:`18458`)
 - Removed :meth:`DataFrame.as_matrix`, :meth:`Series.as_matrix` (:issue:`18458`)
 - Removed :meth:`Series.asobject` (:issue:`18477`)
-- Removed :meth:`DataFrame.as_blocks`, :meth:`Series.as_blocks`, `DataFrame.blocks`, :meth:`Series.blocks` (:issue:`17656`)
+- Removed :meth:`DataFrame.as_blocks`, :meth:`Series.as_blocks`, :attr:`DataFrame.blocks`, :attr:`Series.blocks` (:issue:`17656`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.0.0.rst]: i don't think these work on removed functions / attrivutes

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/whatsnew/v1.0.0.rst (hunk 1 above)
Comment (jschendel) [reply doc/source/whatsnew/v1.0.0.rst]: All of the other headings are `_1000` so making these consistent

Comment (jreback) [reply doc/source/whatsnew/v1.0.0.rst]: hmm, i think they should all be 100, right?

Comment (jschendel) [reply doc/source/whatsnew/v1.0.0.rst]: Ah, yeah, the four digits are probably carry over from when we did "0250" for 0.25.0.  Will switch back to 100 and convert the 1000's.

File: doc/source/whatsnew/v1.0.0.rst (hunk 2 above)
Comment (jschendel) [reply doc/source/whatsnew/v1.0.0.rst]: Yeah, they don't.  This is true for most of the references in this section.

Comment (jreback) [reply doc/source/whatsnew/v1.0.0.rst]: yep, might as well remove the references (but prob doesn't hurt much either)

Comment (jschendel) [reply doc/source/whatsnew/v1.0.0.rst]: okay, went through the entire section and deleted the references from from things that have been removed

**Review by jreback — APPROVED** [review]

**Review by jreback — APPROVED** [review]
lgtm. merge on green.

File: doc/source/whatsnew/v1.0.0.rst (hunk 1 above)
Comment (jschendel) [reply doc/source/whatsnew/v1.0.0.rst]: I guess maybe one argument for keeping 1000 is if we intend to go up to 1.10.0 or beyond.  Probably not a big issue either way.

**Comment by jorisvandenbossche** [comment]
Thanks!

**Pull request merged**